clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -961,11 +962,23 @@ class GCEUtil {
           } else {
             throw jsonException
           }
+          lastSeenException = jsonException
         } catch (GoogleOperationTimedOutException __) {
           log.warn "Retry $action timed out again, trying again..."
         }
       }
-      throw new GoogleOperationException("Failed to $action $resource after #$tries.")
+
+      def lastSeenError = lastSeenException?.getDetails()?.getErrors()[0] ?: null
+      if (lastSeenException && lastSeenError) {
+        throw new GoogleOperationException("Failed to $action $resource after #$tries."
+        + " Last seen exception has status code ${lastSeenException.getStatusCode()} with error message ${lastSeenError.getMessage()}"
+        + " and reason ${lastSeenError.getReason()}.")
+      } else if (lastSeenException) {
+        throw new GoogleOperationException("Failed to $action $resource after #$tries."
+          + " Last seen exception has status code ${lastSeenException.getStatusCode()} with message ${lastSeenException.getMessage()}.")
+      } else {
+        throw new GoogleOperationException("Failed to $action $resource after #$tries.")
```

Comment:
This is currently reachable if we timeout 10 times. It would be better to also catch the retry exceptions and report them as well.